Description: Explicitely link test_map module against eina.
 The module uses eina_stringshare_add() but the build system does not add
 the resective dependency.
Author: Andreas Metzler <ametzler@debian.org>
Origin: vendor
Bug: https://phab.enlightenment.org/T1920

diff --git a/configure.ac b/configure.ac
index cc76226..3a05725 100644
--- a/configure.ac
+++ b/configure.ac
@@ -235,6 +235,7 @@ ELM_CHECK_OPTION([eweb])
 
 PKG_CHECK_MODULES([ELEMENTARY], [${requirement_elm_pc}])
 PKG_CHECK_MODULES([ELEMENTARY_PC], [${requirement_elm_pc_only}])
+PKG_CHECK_MODULES([EINA], [eina  >= efl_version])
 
 EFL_ENABLE_EO_API_SUPPORT
 EFL_ENABLE_BETA_API_SUPPORT
diff --git a/src/modules/test_map/Makefile.am b/src/modules/test_map/Makefile.am
index cf52653..009bbb5 100644
--- a/src/modules/test_map/Makefile.am
+++ b/src/modules/test_map/Makefile.am
@@ -10,6 +10,7 @@ AM_CPPFLAGS = \
 -DPACKAGE_DATA_DIR=\"$(pkgdatadir)\" \
 -DPACKAGE_LIB_DIR=\"$(libdir)\" \
 -DELEMENTARY_BUILD \
+@EINA_CFLAGS@ \
 @ELEMENTARY_CFLAGS@
 
 pkgdir = $(pkglibdir)/modules/test_map/$(MODULE_ARCH)
@@ -17,6 +18,7 @@ pkg_LTLIBRARIES = module.la
 
 module_la_SOURCES = mod.c
 
-module_la_LIBADD = $(top_builddir)/src/lib/libelementary.la
+module_la_LIBADD = $(top_builddir)/src/lib/libelementary.la \
+		   @EINA_LIBS@
 module_la_LDFLAGS = -module @EFL_LTMODULE_FLAGS@
 module_la_LIBTOOLFLAGS = --tag=disable-static
